cloudinit/cmd/main.py

```diff
@@ -1366,8 +1366,14 @@ def all_stages(parser):
 
 def sub_main(args):
 
-    # Subparsers.required = True and each subparser sets action=(name, functor)
-    (name, functor) = args.action
+    try:
+        # Subparsers.required = True
+        # and each subparser sets action=(name, functor)
+        (name, functor) = args.action
+    except AttributeError:
+        print('No Subcommand specified. Please specify a subcommand '
```

Comment:
Let's write this to stderr instead of stdout as it's an error message.
Also let's add a parser.print_usage() before it to give the full expected CLI call usage as a hint.
Finally, let's add a unit test covering this.